iter/counter.go

```diff
@@ -23,4 +27,10 @@ func (c *CountIter) Next() option.Option[int] {
 	return option.Some(c.index - 1)
 }
 
+// String implements the [fmt.Stringer] interface
+func (c *CountIter) String() string {
```

Comment:
Use a non-pointer and then the string representation works for pointers and non-pointers.

```go
func (c CountIter) String() string {
  return "Iterator<Count>"
}
```

Then you can test that the string representations work for both `CountIter` and `*CountIter`

Reply:
Ah, I see, it makes sense.

Reply:
Fixed.